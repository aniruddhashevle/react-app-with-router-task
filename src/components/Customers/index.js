import React from 'react';
import MenuDetailsCards from '../MenuDetailsCards';
import { CUSTOMER_DETAILS } from '../../config/menu-details';

const Customers = () =>
    CUSTOMER_DETAILS && CUSTOMER_DETAILS.length &&
    CUSTOMER_DETAILS.map(item =>
        <MenuDetailsCards key={item.id} config={item} />
    )

export default Customers;
