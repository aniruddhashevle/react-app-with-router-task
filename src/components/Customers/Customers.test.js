import React from 'react';
import Customers from '../Customers';

it('Customers renders correctly', () => {
  const wrapper = shallow(
    <Customers />
  )
  expect(wrapper).toMatchSnapshot();
});