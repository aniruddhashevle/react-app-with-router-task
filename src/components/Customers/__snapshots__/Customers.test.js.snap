// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customers renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Customers />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1",
    "nodeType": "function",
    "props": Object {
      "config": Object {
        "details": "You have 30 customers in last 30 days.",
        "id": 1,
        "pinned": true,
        "title": "Total Customers in November?",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "function",
      "props": Object {
        "config": Object {
          "details": "You have 30 customers in last 30 days.",
          "id": 1,
          "pinned": true,
          "title": "Total Customers in November?",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "2",
      "nodeType": "function",
      "props": Object {
        "config": Object {
          "details": "You have 5 customers in this week.",
          "id": 2,
          "pinned": true,
          "title": "Total Customers this week",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
